packages/caravan-fees/src/cpfp.ts

```diff
@@ -0,0 +1,188 @@
+  if (!parentOutput) {
+    throw new Error(
+      `Spendable output at index ${spendableOutputIndex} not found in the parent transaction.`,
+    );
+  }
+
+  const parentOutputUTXO = availableInputs.find(
+    (utxo) => utxo.txid === analysis.txid && utxo.vout === spendableOutputIndex,
+  );
+
+  if (!parentOutputUTXO) {
+    throw new Error(
+      `UTXO for the spendable output (${analysis.txid}:${spendableOutputIndex}) not found in availableInputs.`,
+    );
+  }
+
+  childTxTemplate.addInput(
+    new BtcTxInputTemplate({
+      txid: analysis.txid,
+      vout: spendableOutputIndex,
+      amountSats: parentOutputUTXO.value,
+    }),
+  );
+
+  // Step 5: Add a change output (required for a valid transaction)
+  childTxTemplate.addOutput(
+    new BtcTxOutputTemplate({
+      address: changeAddress,
+      amountSats: 0, // Initial amount, will be adjusted later
+      type: TxOutputType.CHANGE,
+    }),
+  );
+
+  // Step 6: Add additional inputs if necessary to cover the fee
+  for (const utxo of availableInputs) {
+    if (childTxTemplate.feeRateSatisfied && childTxTemplate.areFeesPayPaid()) {
+      break; // Stop adding inputs once fee requirements are met
+    }
+    // Skip if this UTXO is already added (parent output) or is the parent transaction
+    if (utxo.txid === analysis.txid) {
+      continue;
+    }
+    childTxTemplate.addInput(
+      new BtcTxInputTemplate({
+        txid: utxo.txid,
+        vout: utxo.vout,
+        amountSats: utxo.value,
+      }),
+    );
+  }
+
+  // Step 7: Calculate and set the change output amount
+  const totalInputAmount = childTxTemplate.getTotalInputAmount();
+  const fee = childTxTemplate.targetFeesToPay;
+  const changeAmount = totalInputAmount - fee;
+
+  if (changeAmount <= Number(txAnalyzer.getDustThreshold)) {
+    throw new Error(
+      "Change amount is below the dust threshold. Increase inputs or reduce fee rate.",
+    );
+  }
+
+  childTxTemplate.outputs[0].setAmount(changeAmount);
+
+  // Step 8: Validate the child transaction
+  if (!childTxTemplate.validate()) {
+    throw new Error(
+      "Failed to create a valid CPFP transaction. Ensure all inputs and outputs are valid and fee requirements are met.",
+    );
+  }
+
+  // Step 9: Ensure the combined (parent + child) fee rate meets the target
+  const parentSize = analysis.vsize;
+  const childSize = childTxTemplate.estimatedVsize;
+  const parentFee = parseInt(analysis.fee);
+  const childFee = childTxTemplate.currentFee;
+  const combinedFeeRate =
+    Number(parentFee + childFee) / (parentSize + childSize);
+
+  if (combinedFeeRate < Number(txAnalyzer.cpfpFeeRate)) {
+    throw new Error(
+      `Combined fee rate (${combinedFeeRate.toFixed(2)} sat/vB) is below the target CPFP fee rate (${txAnalyzer.cpfpFeeRate} sat/vB). Increase inputs or reduce fee rate.`,
+    );
+  }
+
+  // Step 10: Convert to PSBT and return as base64
+  const psbt = childTxTemplate.toPsbt();
+  return psbt.serialize("base64");
```

Comment:
yeah, great example here. With my comments above on the final methods on the template this could just be:

```suggestion
  return childTxTemplate.toPsbt();
```

There doesn't seem to be a need to return a `PsbtV2` object at all.

Reply:
Yes changes done [now ](https://github.com/caravan-bitcoin/caravan/pull/114/commits/a215dea59e6ac918e960757d5dd6f74ed09ae55d)..